### PR TITLE
vscode-extensions.sainnhe.gruvbox-material: init at 6.5.2

### LINKED ISSUE #325345: Package request: vscode-extensions.sainnhe.gruvbox-material-vscode
**Project description**

Gruvbox Material color theme for Visual Studio Code 

**Metadata**

* homepage URL: https://github.com/sainnhe/gruvbox-material-vscode
* source URL: https://github.com/sainnhe/gruvbox-material-vscode
* license: mit
* platforms: vscode

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by thtrf**
fixes https://github.com/NixOS/nixpkgs/issues/325345

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by FliegendeWurst — APPROVED** [review]

File: pkgs/applications/editors/vscode/extensions/default.nix [hunk 1]
```diff
@@ -4133,6 +4133,21 @@ let
...
 
+      sainnhe.gruvbox-material = buildVscodeMarketplaceExtension {
+        mktplcRef = {
+          name = "gruvbox-material";
+          publisher = "sainnhe";
+          version = "6.5.2";
+          hash = "sha256-D+SZEQQwjZeuyENOYBJGn8tqS3cJiWbEkmEqhNRY/i4=";
+        };
+        meta = {
+          description = "Gruvbox with Material Palette";
```
Comment (dotlambda) [inline pkgs/applications/editors/vscode/extensions/default.nix:4144]: This doesn't say it's a VSCode extension. Please read the README in this directory.

**Review by dotlambda — COMMENTED** [review]
`meta.changelog` and `meta.maintainers` missing

**Comment by thtrf** [comment]
> `meta.changelog` and `meta.maintainers` missing

I added `meta.changelog`, but `meta.maintainers` is optional.

**Comment by dotlambda** [comment]
> but `meta.maintainers` is optional

I disagree with the README on that, see `pkgs/README.md`.

**Comment by thtrf** [comment]
> > but `meta.maintainers` is optional
> 
> I disagree with the README on that, see `pkgs/README.md`.

Don't you think it's a bit overkill for a vscode theme?

**Comment by dotlambda** [comment]
> Don't you think it's a bit overkill for a vscode theme?

No. It's very useful to have @r-ryantm ping a maintainer who can approve the PR.

**Comment by thtrf** [comment]
> No. It's very useful to have @r-ryantm ping a maintainer who can approve the PR.

I have now added myself as a maintainer.

**Review by FliegendeWurst — APPROVED** [review]

**Pull request merged**